File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minIncrementForUnique = function(nums) {
    nums.sort((a,b)=>a-b)
    let counter=0
    let i=0
    while(i<nums.length-1){
        if(nums[i]>=nums[i+1]){
            counter+=nums[i]-nums[i+1]+1
            nums[i+1]+=nums[i]-nums[i+1]+1
        }
        i++
    }
    return counter
};

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        nums.sort()
        counter=0
        i=0
        while i<len(nums)-1:
            if nums[i]>=nums[i+1]:
                counter+=nums[i]-nums[i+1]+1
                nums[i+1]+=nums[i]-nums[i+1]+1
            i+=1
        return counter